cmake/modules/grpc.cmake

```diff
@@ -3,30 +3,55 @@ option(USE_BUNDLED_GRPC "Enable building of the bundled grpc" ${USE_BUNDLED_DEPS
 if(GRPC_INCLUDE)
 	# we already have grpc
 elseif(NOT USE_BUNDLED_GRPC)
-	find_library(GPR_LIB NAMES gpr)
-	if(GPR_LIB)
-		message(STATUS "Found gpr lib: ${GPR_LIB}")
-	else()
-		message(FATAL_ERROR "Couldn't find system gpr")
-	endif()
-	find_path(GRPCXX_INCLUDE NAMES grpc++/grpc++.h)
-	if(GRPCXX_INCLUDE)
-		set(GRPC_INCLUDE ${GRPCXX_INCLUDE})
-	else()
-		find_path(GRPCPP_INCLUDE NAMES grpcpp/grpcpp.h)
-		set(GRPC_INCLUDE ${GRPCPP_INCLUDE})
-		add_definitions(-DGRPC_INCLUDE_IS_GRPCPP=1)
-	endif()
-	find_library(GRPC_LIB NAMES grpc)
-	find_library(GRPCPP_LIB NAMES grpc++)
-	if(GRPC_INCLUDE AND GRPC_LIB AND GRPCPP_LIB)
-		message(STATUS "Found grpc: include: ${GRPC_INCLUDE}, C lib: ${GRPC_LIB}, C++ lib: ${GRPCPP_LIB}")
+	# gRPC
+	find_package(gRPC CONFIG)
+	if(gRPC_FOUND)
+		message(STATUS "Using gRPC ${gRPC_VERSION}")
+		set(GPR_LIB gRPC::gpr)
+		set(GRPC_LIB gRPC::grpc)
+		set(GRPCPP_LIB gRPC::grpc++)
```

Comment:
What does this do? Does it want to link with `-lgRPC::gpr` or is it some cmake syntax I'm not aware of that pulls the values from the `find_package` result?

Reply:
`gRPC::*` are namespaced targets names exported by `find_package` (those targets are usually defined in `/usr/lib/cmake/grpc/gRPCTargets.cmake`). They can be used as _items_ in `target_link_libraries` and are automatically resolved.

To be more precise, the  [documentation](https://cmake.org/cmake/help/latest/command/target_link_libraries.html#id3) says that an _item_ can be:

> A library target name: The generated link line will have the full path to the linkable library file associated with the target. The buildsystem will have a dependency to re-link <target> if the library file changes.

Regarding the namespaced targets also adds:
> Items containing `::`, such as `Foo::Bar`, are assumed to be IMPORTED or ALIAS library target names and will cause an error if no such target exists.